pixiu/pkg/client/triple/triple.go

```diff
@@ -68,40 +76,47 @@ func (tc *Client) MapParams(req *client.Request) (reqData interface{}, err error
 }
 
 // Close clear GenericServicePool.
-func (dc *Client) Close() error {
+func (tc *Client) Close() error {
 	return nil
 }
 
 // SingletonTripleClient singleton dubbo clent
-func SingletonTripleClient() *Client {
-	return NewTripleClient()
+func SingletonTripleClient(protoset []string) *Client {
+	return NewTripleClient(protoset)
 }
 
 // Call invoke service
-func (dc *Client) Call(req *client.Request) (res interface{}, err error) {
+func (tc *Client) Call(req *client.Request) (res interface{}, err error) {
 	address := strings.Split(req.API.IntegrationRequest.HTTPBackendConfig.URL, ":")
-	p := proxy.NewProxy()
+	p := proxy.NewProxy(tc.protosetSource)
 	targetURL := &url.URL{
 		Scheme: address[0],
 		Opaque: address[1],
 	}
 	if err := p.Connect(context.Background(), targetURL); err != nil {
 		return "", errors.Errorf("connect triple server error = %s", err)
 	}
+
+	header := tc.forwardHeaders(req.IngressRequest.Header)
+	ctx := metadata.NewOutgoingContext(context.Background(), header)
 	meta := make(map[string][]string)
 	reqData, _ := io.ReadAll(req.IngressRequest.Body)
```

Comment:
Do not ignore the error if the request body cannot be read, handle it or throw it.